python/tk_multi_workfiles/work_files.py

```diff
@@ -1050,3 +1058,66 @@ def _get_app_settings_for_context(self, context):
             for settings in engine_settings:
                 if settings.get("app_instance") == app_instance_name:
                     return settings.get("settings")
+
+
+def check_is_rez_compatible(new_ctx):
+    """verify that the given ctx is "compatible" with the current resolved rez environment.
+    Effectively: one shot/entity may have different packages version configured than another one..
+    :param new_ctx: The newly selected context.
+    :return: True if we either are not under Rez, or the new context is ok with the current one.
+    Else: Returns False.
+    """
+    rez_used_request = os.getenv('REZ_USED_REQUEST')
+    if not rez_used_request:
+        # not under Rez so it's always "ok"..
+        return True
+
+    sorted_prev_rez_used_request = sorted(rez_used_request.split())
+
+    # get the "rodeo facility" project :
+    rodeo_facility = new_ctx.sgtk.shotgun.find_one(
+        'CustomNonProjectEntity02', [['code', 'is', 'All rodeo']], fields=['code'])
+
+    # delayed import on purpose, because rdo_rez_launcher isn't available outside of rez ..
+    import rdo_rez_launcher.workflow
+
+    cur_app = os.environ['RDO_CURRENT_APP']
+    cur_show = os.environ['RDO_CURRENT_SHOW']
+
+    workflow = rdo_rez_launcher.workflow.RdoRezLauncherWorkflow(
+        new_ctx.sgtk.shotgun, rodeo_facility,
+        app=cur_app,
+        projectCode=cur_show,
+        entityCode=new_ctx.entity['name'],
+        tankConfigName=new_ctx.sgtk.configuration_name)
+
+    new_ctx_packages = workflow.getCompoundedRezPackagesDict(cur_app)
+    sorted_new_ctx_rez_used_request = sorted('%s%s' % (k, v) for k, v in new_ctx_packages.items())
+
+    if sorted_prev_rez_used_request == sorted_new_ctx_rez_used_request:
+        # both entity have same packages, so fine to continue in this same process
+        return True
+
+    msg = (
+        "The new shot or entity you selected is configured with different version(s) or different package(s) "
+        "than the previous one.\n\n"
+        "This means we have to launch a new instance of {app} to handle the new different context..\n\n"
+        "Is it ok to you ?"
+    ).format(app=cur_app)
+
+    res = QtGui.QMessageBox.question(
+        None, "Rez incompatible contexts ..", msg,
+        QtGui.QMessageBox.Yes | QtGui.QMessageBox.No)
+
+    if res == QtGui.QMessageBox.Yes:
+        cmd_args = [
+            os.environ['RDO_REZ_LAUNCHER_PATH'],  # given by rdo_rez_launcher itself..
+            cur_show,
+            new_ctx.entity['name'],
```

Comment:
If they have selected an entity (shot or asset) then it's there. If they don't (cancel or I don't know) : then this code path isn't called - from what I could see..